Fix ListView css specificity

reidbarber opened the pull request:
When working on ListView's docs, I noticed some css specificity issues that weren't apparent in storybook. This PR essentially moves some selectors to be explicit descendants:

```scss
.react-spectrum-ListViewItem {
  ....
}

.react-spectrum-ListViewItem-checkbox {
  ...
}
```
to
```scss
.react-spectrum-ListViewItem {
  ....
  .react-spectrum-ListViewItem-checkbox {
    ...
  }
}
```

Generic checkbox styles were getting precedence over ListViewItem's checkbox styles in the docs, so this fixes that.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Verify that there are no style changes in storybook

## 🧢 Your Project:

RSP

ktabors commented:
Weird, no generated links
docs - https://reactspectrum.blob.core.windows.net/reactspectrum/eafc90909936a3fecf85a992c91eb02eeed7e02f/docs/index.html
storybook - https://reactspectrum.blob.core.windows.net/reactspectrum/eafc90909936a3fecf85a992c91eb02eeed7e02f/storybook/index.html?path=/story/accordion--default

ktabors reviewed (commented):
Is there a way to test this with your docs?

Otherwise this LGTM in storybook.

- listview--parent-link-example: selection on the row with the link seems buggy. existing issue.

snowystinger reviewed (commented):
How did so many things move around? Asking because most of these blocks look the same, but it's difficult to tell what, if anything has changed on purpose vs potentially accidentally.
is there any way we can move fewer things?

reidbarber commented:
> Is there a way to test this with your docs?

I guess I could make a temp branch off the docs branch, and merge this one into that. Didn't want this change in PR since it's not docs, yet this make its hard to see the fix since we don't currently have any in docs to see.

reidbarber commented:
> How did so many things move around? Asking because most of these blocks look the same, but it's difficult to tell what, if anything has changed on purpose vs potentially accidentally. is there any way we can move fewer things?

So I moved every selector for a descendent of ListViewItem, to be inside ListViewItem (more than just checkbox that I mentioned in the description, also includes image, description, content, etc.). 

While I was there, also cleaned up:

```scss
.react-spectrum-ListView--compact .react-spectrum-ListViewItem {
  ...
}
.react-spectrum-ListView--spacious .react-spectrum-ListViewItem {
  ...
}
```

to

```scss
.react-spectrum-ListViewItem {
  ...

  .react-spectrum-ListView--compact & {
    ...
  }

  .react-spectrum-ListView--spacious & {
    ...
  }
}
```

Sorry, the diff makes it look way more complex than it is.

snowystinger commented on packages/@react-spectrum/list/src/listview.css:
```diff
@@ -97,6 +97,82 @@
+  .react-spectrum-ListViewItem-parentIndicator {
+    grid-area: chevron;
+  
+    [dir="ltr"] & {
+      padding-left: var(--spectrum-global-dimension-size-75);
+    }
+    [dir="rtl"] & {
+      padding-right: var(--spectrum-global-dimension-size-75);
+    }
```
this can actually be condensed to
```suggestion
    padding-inline-start: var(--spectrum-global-dimension-size-75);
```

snowystinger commented on packages/@react-spectrum/list/src/listview.css:
```diff
@@ -97,6 +97,82 @@
+    align-items: center;
+    justify-items: center;
+
+    [dir="ltr"] & {
+      padding-right: var(--spectrum-global-dimension-size-100);
+    }
+    [dir="rtl"] & {
+      padding-left: var(--spectrum-global-dimension-size-100);
+    }
```
```suggestion
    padding-inline-end: var(--spectrum-global-dimension-size-100);
```

reidbarber commented:
@snowystinger reverted the density style change mentioned in the comment above, should be a cleaner diff now :)

snowystinger reviewed (commented):
two easy things to address

LFDanLu reviewed (approved):
This seems fine, but I'm not sure that I see the issue in the ListView docs PR? Chromatic build looks fine too: https://www.chromatic.com/build?appId=5f0dd5ad2b5fc10022a2e320&number=198

reidbarber commented:
> This seems fine, but I'm not sure that I see the issue in the ListView docs PR? Chromatic build looks fine too: https://www.chromatic.com/build?appId=5f0dd5ad2b5fc10022a2e320&number=198

Yeah, I'm only seeing the issue locally:

<img width="351" alt="Screen Shot 2022-01-13 at 1 28 28 PM" src="https://user-images.githubusercontent.com/8961049/149396550-1cc9e868-7074-4b27-ac1f-e05b96a74a3c.png">
